src/app/shared/components/comment-form/comment-form.component.ts

```diff
@@ -48,44 +49,61 @@ export class CommentFormComponent implements OnInit {
   ) {}
 
   ngOnInit(): void {
+    this.updateFormType();
     if (this.data.comment != null) {
-      this.title = 'Uredi komentar';
-      this.commentForm.patchValue({ content: this.data.comment.content });
-
-      if (this.data.comment.type === 'warning') {
-        this.addExposedUntilField();
-        this.commentForm.patchValue({
-          exposedUntil: this.data.comment.exposedUntil,
-        });
-      }
+      this.commentForm.patchValue({
+        content: this.data.comment.content,
+        type: this.data.comment.type,
+      });
     }
+  }
 
-    switch (this.data.type) {
+  onTypeChange() {
+    this.updateFormType();
+  }
+
+  updateFormType() {
+    let prefix = this.data.comment != null ? 'Uredi' : 'Dodaj';
+
+    switch (this.type) {
+      case 'comment':
+        this.title = `${prefix} komentar`;
+        this.removeExposedUntilField();
+        break;
       case 'warning':
-        this.title = 'Dodaj opozorilo';
+        this.title = `${prefix} opozorilo`;
         this.addExposedUntilField();
+        if (this.data.comment != null) {
+          this.commentForm.patchValue({
+            exposedUntil: this.data.comment.exposedUntil,
+          });
+        }
         break;
       case 'condition':
-        this.title = 'Dodaj informacijo o razmerah';
-        break;
-      case 'comment':
-        this.title = 'Dodaj komentar';
+        this.title = `${prefix} opis razmer`;
+        this.removeExposedUntilField();
         break;
       case 'description':
         // TODO: it seems that only peaks and routes can have comments of type description. Might want to include entity type in the title and generalize this. Should discuss.
-        this.title = 'Dodaj opis';
+        this.title = `${prefix} opis smeri`;
+        this.removeExposedUntilField();
         break;
-      default:
-        this.title = '';
     }
   }
 
   addExposedUntilField() {
-    this.commentForm.addControl('exposedUntil', new FormControl());
+    this.commentForm.addControl(
+      'exposedUntil',
+      new FormControl(null, [Validators.required])
```

Comment:
Prevent the user from submitting the form without the `exposedUntil` date. The endpoint fails anyway and the displayed message in this case is not helpful.

Reply:
We should let the user post warnings without exposing them on home page. -> so this field should stay not required...
maybe additional explanation (in a hint) would be welcome. or even a checkbox?

why BE fails? probably this is a FE bug that I suspect was introduced when migrating from momentjs to dayjs. dayjs does not return null when null is fed to it...

Reply:
resolved this bug with: https://github.com/plezanje-net/web/pull/402/commits/8e575ae87fdfb77659d1fc737c0a720acaaee670